### PR TITLE
feat(xss-context): add example for context sensitive XSS encoding issues

### PR DESCRIPTION
**Pull request by lirantal**
Adding a case where we use the same variable as a URL for a hyperlink. While the variable is encoded for HTML for the input element, it isn't suitable for the URL entry.

owasp node js goat project context sensitive xss

### CODE REVIEW

**Comment by binarymist** [comment]
Just looking at the file changes alone, I don't see any issues.

**Comment by lirantal** [comment]
Cool, will go ahead and merge.

**Comment by ckarande** [comment]
@lirantal Thanks for this addition. It highlights the importance of context sensitive encoding. I liked your creativity to come up with a way to demonstrate it on the profile page. If you have a few minutes, can you also add a few lines explaining this issue in the Tutorial as well; otherwise it may go unnoticed by the users.

**Comment by lirantal** [comment]
@ckarande yes of course, I'll open a follow-up PR with all the suggested improvements.

File: app/routes/profile.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
             doc.userId = userId;
 
+            // @TODO @FIXME
+            // while the developer intentions were correct in encoding the user supplied input so it
+            // doesn't end up as an XSS attack, the context is incorrect as it is encoding the firstname for HTML
+            // while this same variable is also used in the context of a URL link element
+            doc.firstNameSafeString = ESAPI.encoder().encodeForHTML(doc.firstName)
+            // fix it by replacing the above with another template variable that is used for 
+            // the context of a URL in a link header
+            // doc.firstNameSafeString = ESAPI.encoder().encodeForURL(urlInput)
```
[the remark this answers is not in the document]
Comment (lirantal) [reply app/routes/profile.js]: Yes makes sense!

File: app/views/profile.html [hunk 2]
```diff
@@ -67,6 +67,8 @@ <h3 class="panel-title">Edit Profile</h3>
                     </div>
                     <input type="hidden" name="_csrf" value="{{csrftoken}}" />
                     <button type="submit" class="btn btn-default" name="submit">Submit</button>
+
```
[the remark this answers is not in the document]
Comment (lirantal) [reply app/views/profile.html]: 👍

**Pull request merged**